### PR TITLE
[test](index compaction)Add index compaction exception fault injection cases

### PR DESCRIPTION
**Pull request by qidaye**
### What problem does this PR solve?

Problem Summary:

1. Add skip index compaction logic when index compaction encounters exceptions
2. Optimize fault injection regression cases

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by qidaye** [comment]
run buildall

**Review by zzzxl1993 — APPROVED** [review]
LGTM

File: be/src/olap/compaction.cpp [hunk 1]
```diff
@@ -714,6 +788,16 @@ Status Compaction::do_inverted_index_compaction() {
     return Status::OK();
 }
 
+void Compaction::mark_skip_index_compaction(
+        const RowsetWriterContext& context,
+        const std::function<void(int64_t, int64_t)>& error_handler) {
+    for (auto&& column_uniq_id : context.columns_to_do_index_compaction) {
+        auto col = _cur_tablet_schema->column_by_uid(column_uniq_id);
+        const auto* index_meta = _cur_tablet_schema->inverted_index(col);
```
Comment (csun5285) [inline be/src/olap/compaction.cpp:796]: index_meta may be nullptr ?

Comment (qidaye) [reply be/src/olap/compaction.cpp:796]: Good catch!

**Comment by qidaye** [comment]
run buildall

**Review by csun5285 — APPROVED** [review]
LGTM

**Review by airborne12 — APPROVED** [review]
LGTM

**Pull request merged**